apps/src/Register3D.cpp

```diff
@@ -0,0 +1,185 @@
+#include <fstream>
+#include <iostream>
+
+#include <boost/filesystem.hpp>
+#include <itkCompositeTransform.h>
+#include <itkTransformFileReader.h>
+#include <itkTransformFileWriter.h>
+#include <opencv2/core.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <vc/core/io/OBJReader.hpp>
+#include <vc/core/io/OBJWriter.hpp>
+#include <vc/core/types/Exceptions.hpp>
+#include <vc/core/types/UVMap.hpp>
+
+#include "rt/DeformableRegistration.hpp"
+#include "rt/ImageTransformResampler.hpp"
+#include "rt/ImageTypes.hpp"
+#include "rt/LandmarkIO.hpp"
+#include "rt/LandmarkRegistration.hpp"
+#include "rt/itk/itkOpenCVImageBridge.h"
+
+namespace fs = boost::filesystem;
+
+// Composite Transform
+using CompositeTransform = itk::CompositeTransform<double, 2>;
+
+// IO
+using TransformWriter = itk::TransformFileWriterTemplate<double>;
+using TransformReader = itk::TransformFileReaderTemplate<double>;
+
+int main(int argc, char* argv[])
+{
+    if (argc < 7) {
+        std::cerr << "Missing Parameters " << std::endl;
+        std::cerr << "Usage: " << argv[0];
+        std::cerr << " objFile landmarksFile ";
+        std::cerr << "movingImage iterations";
+        std::cerr << "objOutputFileName ";
+        std::cerr << "transformOutputFileName " << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    auto objFileName = argv[1];
```

Comment:
Use fs::path instead of auto here. In this circumstance, the compiler will likely deduce the type to be a char* or a string. I may have messed this up in register2d.

Also, call it `objPath`. Prefer `path` to `filename`. It's shorter and just as descriptive.

Do the same for all of the other path arguments below.